rust_src/src/ng_async.rs

```diff
@@ -239,14 +357,58 @@ pub async fn async_echo(s: String) -> String {
     s
 }
 
+#[async_stream]
+pub async fn async_data_echo(e: UserData) -> UserData {
+    e
+}
+
+fn internal_send_message(
+    pipe: &mut EmacsPipe,
+    message: LispObject,
+    option: PipeDataOption,
+) -> bool {
+    match option {
+        PipeDataOption::STRING => {
+            if !message.is_string() {
+                wrong_type!(Qstringp, message);
+            }
+
+            let encoded_message = unsafe { encode_string_utf_8(message, Qnil, false, Qt, Qt) };
+            let encoded_string: LispStringRef = encoded_message.into();
+            let contents = String::from_utf8_lossy(encoded_string.as_slice());
```

Comment:
Now that we are explicitly encoding this with emacs implementation of utf8_encode, this is a candidate for String::from_utf8_unchecked. That function in unsafe because it does not perform a utf8 validity check. I have not done any research on the correctness of emacs `encode_string_utf_8` impl, nor have I benchmarked the performance between utf8_lossy and utf8_unchecked. Due to these reasons, I am leaving this as a call to `from_utf8_lossy` for now

Reply:
Fine with me, we can change this later.